src/hooks/useLocalAndRemotePatients.js

```diff
@@ -0,0 +1,153 @@
+import { useEffect, useReducer } from 'react';
```

Comment:
Aaaand this is where it gets a bit hairy.

I started off using thunks and redux to do the API calls.

It was getting pretty messy, knowing when data was coming from APIs or from realm, to know when to filter. It was also hard to track loading/error states through redux.

I thought to use the `useFetch` hook which handles most of this already.

What I ended up with was this hook which is an attempt to create an interface where data from two different sources are kind of merged to form a single source of state.

There will also need to be pagination logic added to this.

It's a bit complex but I think it is less so then mixing hooks and different redux stuff going on